Opentrons/opentrons
# refactor(robot-server,app): Use tiprack from protocol in tip length calibration

Tip length calibration must be run on the same tiprack that is used in the protocol. It also in the future will have to be run on either a standard or user-selectable tiprack to establish tip length before instrument offset; and in the farther future, tip length calibration may be runnable at will outside of the protocol flow.

To support this,
- The tip rack definition to calibrate is now sent as a `createParam` to the tip length calibration session
- The app loads the tip rack definition for the tip rack to calibrate using the same mechanisms it uses for displaying labware. This unfortunately means #6234 affects the app as well, but it handles custom labware sourced in json protocols or protocols uploaded from the local app acceptably.

## Completion

This PR is complete and addresses both the app and server sides. The one thing that is missing is a placeholder image for tipracks that we do not have images for.

## Testing
- Upload protocols using non-standard tips
- Check that the introduction screen shows the appropriate image and displays the appropriate display name
- Check that the deck layout displays the appropriate load name

Closes #6232
